src/sound.cpp

```diff
@@ -233,6 +233,34 @@ void SndCopyToPool()
 	}
 }
 
+/**
+ * Change the configured sound set and reset sounds.
+ * @param index Index of sound set to switch to.
+ */
+void ChangeSoundSet(int index)
+{
+	if (BaseSounds::GetIndexOfUsedSet() == index) return;
+
+	auto set = BaseSounds::GetSet(index);
+	BaseSounds::ini_set = set->name;
+	BaseSounds::SetSet(set);
+
+	MxCloseAllChannels();
+	InitializeSound();
+
+	/* Replace baseset sounds in the pool with the updated original sounds. This is safe to do as
+	 * any sound still playing holds its own shared_ptr to the sample data. */
```

Comment:
Do you actually need to close channels if it's using `shared_ptr` for the sample data? Can't the samples just remain as is. So you'll let the "old" sounds run out and start "new" sounds. Only if it's simpler though.

Reply:
Yes, they can continue playing, it's more a user-expectation thing. If they aren't stopped and carry on playing, it can appear like the change has had no effect.